daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -307,50 +306,14 @@ object SExpr {
     }
   }
 
-  /** A non-recursive, non-parallel let block. Each bound expression
-    * is evaluated in turn and pushed into the environment one by one,
-    * with later expressions possibly referring to earlier.
+  /** A non-recursive, non-parallel let block.
+    * It is used as an intermediary data structure by the compiler to
+    * mitigate stack overflow issues, but are later exploded into
+    * [[SELet1General]] and [[SELet1Builtin]] by the ANF transformation.
     */
-  final case class SELet(bounds: Array[SExpr], body: SExpr) extends SExpr with SomeArrayEquals {
-    def execute(machine: Machine): Unit = {
-
-      // Evaluate the body after we've evaluated the binders
-      machine.pushKont(
-        KPushTo(
-          machine.env,
-          body,
-          machine.frame,
-          machine.actuals,
-          machine.env.size + bounds.size - 1))
-
-      // Start evaluating the let binders
-      var i = 1
-      while (i < bounds.size) {
-        val b = bounds(bounds.size - i)
-        val expectedEnvSize = machine.env.size + bounds.size - i - 1
-        machine.pushKont(KPushTo(machine.env, b, machine.frame, machine.actuals, expectedEnvSize))
-        i += 1
-      }
-      machine.ctrl = bounds.head
-    }
-  }
-
-  object SELet {
-
-    // Helpers for constructing let expressions:
-    // Instead of
-    //   SELet(Array(SEVar(4), SEVar(1)), SEVar(1))
-    // you can write:
-    //   SELet(
-    //     SEVar(4),
-    //     SEVar(1)
-    //   ) in SEVar(1)
-    case class PartialSELet(bounds: Array[SExpr]) extends SomeArrayEquals {
-      def in(body: => SExpr): SExpr = SELet(bounds, body)
-    }
-
-    def apply(bounds: SExpr*) = PartialSELet(bounds.toArray)
-
+  final case class SELet(bounds: List[SExpr], body: SExpr) extends SExpr {
+    def execute(machine: Machine): Unit =
+      crash("not implemented")
```

Comment:
"not implemented" sounds to me like we'll implement it in the future.
```suggestion
      crash("unexpected SELet during execution")
```